src/api/app/models/project/update_from_xml_command.rb

```diff
@@ -282,22 +282,28 @@ def update_hostsystem(current_repo, xml_hash)
       current_repo.save! if current_repo.changed?
     end
 
-    def update_repository_architectures(current_repo, xml_hash)
-      # destroy architecture references
-      Rails.logger.debug "delete all repository architectures of repository '#{project.id}'"
-      RepositoryArchitecture.where('repository_id = ?', current_repo.id).delete_all
+    def check_for_duplicated_archs!(xml_archs)
+      taken_archs = {}
```

Comment:
``xml_archs`` is an array I guess? Than this can be simplified with:

```ruby
def check_for_duplicated_archs!(architectures)
  duplicate_architectures = architectures - architectures.uniq 
  return if duplicate_architectures.empty

  raise SaveError, "double use of architecture: '#{duplicate_architectures.first}'"
end
```

Reply:
Argh, it's removing both elements ... :cry: 

What about

```
%w(i586 i586) & %w(i586)
```

Reply:
let me pull out the 'this is a little out of scope for this PR' hammer.

Reply:
the problem is that `%w(i586) & %w(i586)` has the same result without duplicating.